AST StringLiteral node has optional `lvalue` token in Clang 9.0

The `lvalue` token of the `StringLiteral` node is optional  in Clang 9.0.

# Example

Contents of `f.c`:
```c
const char s[] = "foo";
```

# Panic

```
$ c2go transpile f.c
panic: could not match regexp with string
^(?P<address>[0-9a-fx]+) <(?P<position>.*)> '(?P<type>.*)' lvalue (?P<value>".*")[\s]*$
0x555c86cab818 <col:18> 'const char [4]' "foo"


goroutine 52 [running]:
github.com/elliotchance/c2go/ast.groupsFromRegex(0xc00011c660, 0x57, 0xc000110490, 0x2e, 0x40)
	/home/u/goget/src/github.com/elliotchance/c2go/ast/ast.go:290 +0x360
github.com/elliotchance/c2go/ast.parseStringLiteral(0xc000110490, 0x2e, 0x6b45fa)
```

# AST output

The AST of `f.c` looks as follows:

```
$ c2go ast f.c
TranslationUnitDecl 0x5632b5f4a9f8 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x5632b5f4b290 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x5632b5f4af90 '__int128'
|-TypedefDecl 0x5632b5f4b300 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x5632b5f4afb0 'unsigned __int128'
|-TypedefDecl 0x5632b5f4b5e8 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x5632b5f4b3e0 'struct __NSConstantString_tag'
|   `-Record 0x5632b5f4b358 '__NSConstantString_tag'
|-TypedefDecl 0x5632b5f4b680 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x5632b5f4b640 'char *'
|   `-BuiltinType 0x5632b5f4aa90 'char'
|-TypedefDecl 0x5632b5f4b958 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x5632b5f4b900 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x5632b5f4b760 'struct __va_list_tag'
|     `-Record 0x5632b5f4b6d8 '__va_list_tag'
`-VarDecl 0x5632b5fa9728 </home/u/Desktop/ccc/f/f.c:1:1, col:18> col:12 s 'const char [4]' cinit
  `-StringLiteral 0x5632b5fa9818 <col:18> 'const char [4]' "foo"
```

--- a/ast/string_literal.go
+++ b/ast/string_literal.go
@@ -17,7 +17,7 @@ type StringLiteral struct {
 
 func parseStringLiteral(line string) *StringLiteral {
 	groups := groupsFromRegex(
-		`<(?P<position>.*)> '(?P<type>.*)' lvalue (?P<value>".*")`,
+		`<(?P<position>.*)> '(?P<type>.*)'(?P<lvalue> lvalue)? (?P<value>".*")`,
 		line,
 	)
 
@@ -31,7 +31,7 @@ func parseStringLiteral(line string) *StringLiteral {
 		Pos:        NewPositionFromString(groups["position"]),
 		Type:       groups["type"],
 		Value:      s,
-		Lvalue:     true,
+		Lvalue:     len(groups["lvalue"]) > 0,
 		ChildNodes: []Node{},
 	}
 }

--- a/ast/string_literal_test.go
+++ b/ast/string_literal_test.go
@@ -22,6 +22,14 @@ func TestStringLiteral(t *testing.T) {
 			Value:      "x\vx\x00xxx\axx\tx\n",
 			ChildNodes: []Node{},
 		},
+		`0x5632b5fa9818 <col:18> 'const char [4]' "foo"`: &StringLiteral{
+			Addr:       0x5632b5fa9818,
+			Pos:        NewPositionFromString("col:18"),
+			Type:       "const char [4]",
+			Lvalue:     false,
+			Value:      "foo",
+			ChildNodes: []Node{},
+		},
 	}
 
 	runNodeTests(t, nodes)

